dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -784,6 +785,16 @@ public void checkSerialProcess(ProcessDefinition processDefinition) {
             return;
         }
         Map<String, Object> cmdParam = new HashMap<>();
+        // write the parameters of the nextProcessInstance to command
+        if (StringUtils.isNotEmpty(nextProcessInstance.getCommandParam())) {
+            Map<String, String> commandStartParamsMap = JSONUtils.toMap(nextProcessInstance.getCommandParam());
+            if (MapUtils.isNotEmpty(commandStartParamsMap)) {
+                Map<String, String> paramsMap = JSONUtils.toMap(commandStartParamsMap.get(CMD_PARAM_START_PARAMS));
+                if (MapUtils.isNotEmpty(paramsMap)) {
+                    cmdParam.put(CMD_PARAM_START_PARAMS, JSONUtils.toJsonString(paramsMap));
+                }
+            }
+        }
```

Comment:
Why we need to set the workflow instance param to command?

Reply:
The next process instance to be awakened will create a new command, and The new command needs to keep the original "StartParams" fields

Reply:
Whether I need to change the code   @ruanwenjun

Reply:
It should work, but it's better to change the command consumer logic rather than put the param in wakeup command.

Reply:
> It should work, but it's better to change the command consumer logic rather than put the param in wakeup command.

thanks!